### ISSUE TITLE
QuickAccess on Remote Development is slow

### ISSUE THREAD

**Issue by hullis** [microsoft/vscode#157056]
Issue Type: <b>Bug</b>

I don't now how to reproduce it on other machines. On my occasion I am running vscode remote on a Linux (Debian) machine. So I am going to upload a profile here and see if it helps.

<img width="1133" alt="截屏2022-08-04 10 06 28" src="https://user-images.githubusercontent.com/12122021/182748197-d8017a89-c1d8-4f7a-abe5-30ec6bbcea89.png">

<img width="1052" alt="截屏2022-08-04 10 13 23" src="https://user-images.githubusercontent.com/12122021/182748221-644700fe-c18a-4281-89f5-5a061a144b05.png">

It seems that it is related to `findFormatting`.

VS Code version: Code 1.69.2 (3b889b090b5ad5793f524b5d1d39fda662b96a2a, 2022-07-18T16:12:57.074Z)
OS version: Darwin x64 21.6.0
Restricted Mode: No
Remote OS version: Linux x64 5.4.56.bsk.10-amd64

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|32.00GB (5.18GB free)|
|Process Argv|--crash-reporter-id 6f7fef97-bf68-4304-98b4-8812ae5d3ad3|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: devcloud|
|OS|Linux x64 5.4.56.bsk.10-amd64|
|CPUs|Intel(R) Xeon(R) Platinum 8260 CPU @ 2.40GHz (8 x 2394)|
|Memory (System)|15.01GB (8.55GB free)|
|VM|0%|
</details><details><summary>Extensions (18)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|13.3.1
remote-ssh|ms-|0.85.2022071315
remote-ssh-edit|ms-|0.80.0
svg-preview|Sim|2.8.3
vscode-icons|vsc|11.14.0
vim|vsc|1.23.2
Bookmarks|ale|13.3.1
vscode-eslint|dba|2.2.6
gitlens|eam|12.1.2
EditorConfig|Edi|0.16.6
prettier-vscode|esb|9.5.0
todo-tree|Gru|0.0.215
git-graph|mhu|1.30.0
svg-preview|Sim|2.8.3
code-spell-checker|str|2.3.4
tldraw-vscode|tld|1.19.0
vscode-icons|vsc|11.14.0
vitest-explorer|Zix|0.2.23

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonvspyt551:30345470
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscorecescf:30445987
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
vscscmwlcmt:30465135
cppdebug:30492333
pylanb8912cf:30529770
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253

```

</details>

**Comment by hullis** [comment]
[Profile-20220804T100958.json.zip](https://github.com/microsoft/vscode/files/9256160/Profile-20220804T100958.json.zip)

**Comment by TylerLeonhardt** [comment]
Can you describe what you're doing when you see this slowness?

**Comment by hullis** [comment]
<img width="966" alt="截屏2022-08-06 10 28 17" src="https://user-images.githubusercontent.com/12122021/183230026-68815783-e5b1-4127-9ba7-8d7b108858e9.png">

I digged a little bit and foud out there are thousands of formatters. And there are loaded from local storage. @TylerLeonhardt

**Comment by hullis** [comment]
> Can you describe what you're doing when you see this slowness?

I was using <kbd>cm + p</kbd> to search files.

**Comment by hullis** [comment]
https://github.com/microsoft/vscode/blob/cb79d015557d546d0d48cea97d7fc3ac0d3dc590/src/vs/workbench/services/label/common/labelService.ts#L370-L402

Before `registerCachedFormatter` is called, `registerFormatter` gets called serveral times and pushes formatters into `this.formatters`. Since `this.storedFormatters` and `this.formatters` point to the same array.

https://github.com/microsoft/vscode/blob/cb79d015557d546d0d48cea97d7fc3ac0d3dc590/src/vs/workbench/services/label/common/labelService.ts#L147-L148

These newly registered formatters get stored as well.

**Comment by hullis** [comment]
Oh I see it's already fixed #157313. Closing this now.
